merge upstream

maflaven opened the pull request.

msvticket committed:
Make the jvm figure out memory settings from the memory limit of the docker container

(cherry picked from commit b32612b084e79c82f17afd7036df232fcc170eae)

Tazer committed:
fixed comment to be correct for go watch.sh

garethjevans committed:
fix: removed unused detect scripts

garethjevans committed:
feat: added buildpack for CWP

Commit:
Fix javascript pack

jstrachan committed:
fix: add a maven11 build pack

hekonsek committed:
Fixed container name.

hekonsek committed:
Maven11 should use JRE base image instead of SDK.

jstrachan committed:
Merge pull request #16 from hekonsek/maven11-container-fix

Fixed Maven11 container name

jstrachan committed:
Merge pull request #17 from hekonsek/maven11-jre2

Maven11 should use JRE base image instead of SDK

jstrachan committed:
chore: refactored to better name

jstrachan committed:
chore: reuse maven as the container name for all java versions

sebosp committed:
Debian:8 for newer glibc

It is not possible to run rust-http under centos:7 docker image with Rocket 0.4.0 since it requires glibc 2.18 or higher, centos:7 provides glibc2.17 and when the container starts (after build finishes):
`/my-rust-http-test: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /my-rust-http-test)`
By using Debian:8, applications using 0.4.0 rust-http quickstart can be started.

jstrachan committed:
fix: enable batch mode for 'jx step changelog'

jstrachan committed:
fix: add a first spike at a build pack for environments

jstrachan committed:
fix: use a better default container for environment pipelines

jstrachan committed:
revert accidental change

jstrachan committed:
fix: use a better default container for environment pipelines

Commit:
Added go-mongo buildpack

Commit:
Added go-mongo buildpack

igdianov committed:
feat: (Draft) add yaml Draft build pack support for Helm quickstarts

Commit:
Renamed to go-mongodb

igdianov committed:
fix: change pipeline agent to jenkins-maven

jstrachan committed:
fix: lets give steps a simple canonical name

for easier reporting and to make overriding steps easier

jstrachan committed:
Merge pull request #21 from igdianov/master

feat: (Draft) add yaml Draft build pack support for Helm quickstarts

jstrachan committed:
fix: rename the 'yaml' build pack to 'helm'

as most quickstarts have more YAML than anything else so its better to use a non-language name like 'helm' - then if there's no programming language build pack found, we default to 'helm' if we find a `*/Chart.yaml` file

also added a docker build pack

jstrachan committed:
fix: first spike at a jenkins build pack

jstrachan committed:
fix: add skaffold yaml

jstrachan committed:
fix: add a default Dockerfile

jstrachan committed:
fix: lets add a docker and docker helm build pack

jstrachan committed:
fix: pure docker pipeline

zackpayton committed:
modified python build pack to be able to build correctly

jstrachan committed:
fix: upgrade exposecontroller

Commit:
Added go-mongo buildpack

yyolk commented:
@ranginuitrot I noticed `Jenkinsfile`s have been removed in favor of `pipeline.yaml` 🎉

ranginuitrot commented:
@yyolk yeah I've been wanting to have @McRobie look into switching over to yaml for our custom packs too.

jstrachan committed:
fix: add custom jenkins build pack

yyolk reviewed (approved):
i'm fine with all the things in here

pow-devops2020 committed:
fix: PR pipeline for environment build packs

Minipada committed:
feat: add C++ pack

jstrachan committed:
Merge pull request #27 from Minipada/cpp_draft

feat: add C++ pack

warrenbailey committed:
(feat) Chart buildpack initial check in

Creating a new buildpack for the jenkins-x-charts repos

warrenbailey committed:
Merge pull request #28 from warrenbailey/charts-build-pack

(feat) Chart buildpack initial check in

warrenbailey committed:
Jenkins X Apps Build pack

Initial check in of apps build pack

pow-devops2020 committed:
fix: the set version command

pow-devops2020 committed:
fix: use better set version command

jstrachan committed:
Merge pull request #22 from vfarcic/master

Go With MongoDB

Commit:
Removed duplicated chart

jstrachan committed:
Merge pull request #29 from vfarcic/master

Removed duplicated chart

Commit:
Removed password

pow-devops2020 committed:
fix: remove redundant extra tag

jstrachan committed:
Merge pull request #30 from vfarcic/master

Removed password

jstrachan committed:
Merge pull request #24 from westwardai/fix_python_build_pack

modified python build pack to be able to build correctly

jstrachan committed:
Merge pull request #14 from garethjevans/cwp

feat: added buildpack for CWP

jstrachan committed:
Merge pull request #18 from sebosp/patch-1

Debian:8 for newer glibc

jstrachan committed:
Merge pull request #15 from hnicke/fix-javascript-pack

Fix javascript pack

jstrachan committed:
Merge pull request #13 from garethjevans/detect

fix: removed unused detect scripts

jstrachan committed:
Merge pull request #10 from msvticket/master

Make the jvm figure out memory settings from the memory limit of the docker container

jstrachan committed:
Merge pull request #2 from Tazer/comment-correct-draft-pack

fixed comment to be correct for go watch.sh

warrenbailey committed:
Correct indentation

warrenbailey committed:
Added build step

jstrachan committed:
Update pipeline.yaml

jstrachan committed:
Update pipeline.yaml

jstrachan committed:
Update pipeline.yaml

pow-devops2020 committed:
hack to work around current bug in configuring images

warrenbailey committed:
Merge pull request #32 from warrenbailey/apps-build-pack

Apps Build pack

garethjevans committed:
feat: adding a nop build pack that doesn't do anything

garethjevans committed:
Merge pull request #34 from garethjevans/nop

feat: adding a nop build pack that doesn't do anything

garethjevans committed:
fix: configured correct container for nop build

garethjevans committed:
Merge pull request #35 from garethjevans/nopfix

fix: configured correct container for nop build

jstrachan committed:
fix: add optional support for knative serve

also added a consistent way to add environment variables via the `values.yaml` file

also added some missing liveness probes

warrenbailey committed:
Remove build from the promote step

warrenbailey committed:
Merge pull request #37 from warrenbailey/correct-apps-build-pack

Remove build from the promote step

warrenbailey committed:
Add environment variables

warrenbailey committed:
Merge pull request #38 from warrenbailey/environment-variables

App Build Pack Environment variables

jstrachan committed:
fix: allow the service name to be overloaded

to help fix up preview URLs

jstrachan committed:
fix: use a better default container name for environments

rawlingsj committed:
remove individual step containers as they're all 

the same and we can use a top level one

rawlingsj committed:
Merge pull request #41 from jenkins-x-buildpacks/rawlingsj-patch-1

remove individual step containers as they're all

jstrachan committed:
fix: first spike of a php build pack

jstrachan committed:
fix: avoid docker build and use skaffold/kaniko instead

jstrachan committed:
fix: php chart

jstrachan committed:
fix: let default to exposecontroller for php

jstrachan committed:
fix: lets use port 80 for php internal ports

jstrachan committed:
fix: lets add a preview environment to the php build pack

tdcox committed:
feat: Added packs for machine-learning

jstrachan committed:
Merge pull request #42 from tdcox/machine-learning

feat: Added packs for machine-learning

tdcox committed:
fix: Revise PR generation against new helper method

Signed-off-by: Terry Cox <terry@bootstrap.je>

fix: Push as part of creating pull request

Signed-off-by: Terry Cox <terry@bootstrap.je>

fix: Commit as part of creating pull request

Signed-off-by: Terry Cox <terry@bootstrap.je>

fix: Explicitly include title and body

Signed-off-by: Terry Cox <terry@bootstrap.je>

garethjevans committed:
fix: updated DOCKER_REGISTRY for scala builds

garethjevans committed:
Merge pull request #48 from garethjevans/docker-registry

fix: updated DOCKER_REGISTRY for scala builds

garethjevans committed:
feat: use templating for image name

garethjevans committed:
feat: using IMAGE_NAME for tagging

garethjevans committed:
feat: updated typesafe pack

garethjevans committed:
fix: corrected skaffold format

garethjevans committed:
fix: corrected scala buildpack

garethjevans committed:
feat: using REPLACE_ME_DOCKER_REGISTRY_ORG in Makefiles

garethjevans committed:
Merge pull request #47 from tdcox/ml-2

fix: Revise PR generation against new helper method

garethjevans committed:
Merge pull request #49 from garethjevans/changeme

feat: use templating for image name

garethjevans committed:
fix: removed empty artifact block which is incompatible with skaffold v0.29.0

garethjevans committed:
Merge pull request #50 from garethjevans/fix-empty-artifact

fix: removed empty artifact block which is incompatible with skaffold v0.29.0

jwtodd committed:
fixes #51 make knativeDeploy:true work; tested addon:gloo and deployed node-http to gke. note: not sure of the ramifications of removing terminateionGracePeriodSeconds but i am pretty sure it is deprecated

jstrachan committed:
Merge pull request #53 from jwtodd/master

fixes #51 make knativeDeploy:true work; tested addon:gloo and deploye…

tdcox committed:
feat: Add packs for Machine Learning with CUDA

Signed-off-by: Terry Cox <terry@bootstrap.je>

tdcox committed:
feat: Transition to new pipeline format

Signed-off-by: Terry Cox <terry@bootstrap.je>

tdcox committed:
fix: Specify agent in pipelines

Signed-off-by: Terry Cox <terry@bootstrap.je>

tdcox committed:
fix: Specify agent correctly in pipelines

Signed-off-by: Terry Cox <terry@bootstrap.je>

tdcox committed:
fix: Remove comments

Signed-off-by: Terry Cox <terry@bootstrap.je>

tdcox committed:
fix: Remove escapes from sh commands

Signed-off-by: Terry Cox <terry@bootstrap.je>

tdcox committed:
fix: Use recursive copy for directory

Signed-off-by: Terry Cox <terry@bootstrap.je>

jstrachan committed:
Merge pull request #55 from tdcox/ml-6

feat: Add packs for Maching Learning with CUDA

tdcox committed:
fix: typo in pipeline

Signed-off-by: Terry Cox <terry@bootstrap.je>

jstrachan committed:
Merge pull request #57 from tdcox/ml-7

fix: typo in pipeline

tdcox committed:
chore: Update dependencies to include pytorch 1.1.0 and onnx 1.5.0

Signed-off-by: Terry Cox <terry@bootstrap.je>

jstrachan committed:
Merge pull request #58 from tdcox/ml-8

chore: Update dependencies to include pytorch 1.1.0 and onnx 1.5.0

Commit:
fix: Quote the environment variable values in deployment.yaml to avoid the 'cannot convert int64 to string' error when the value is an integer (even quoted) in values.yaml.

jstrachan committed:
Merge pull request #59 from hervelemeur/master

fix: Quote the environment variable values in deployment.yaml to avoid 'cannot convert int64 to string' error

tdcox committed:
fix: Catch situation where pipeline is run again against existing PR

Signed-off-by: Terry Cox <terry@bootstrap.je>

jstrachan committed:
Merge pull request #63 from tdcox/ml-9

fix: Catch situation where pipeline is run again against existing PR

VincentS committed:
Fixed typo in Chart.yaml 

The golang Helm Chart had `javascript` instead of `golang` set as name.

warrenbailey committed:
Merge pull request #64 from VincentS/patch-1

Fixed typo in Chart.yaml

jstrachan committed:
fix: add a git build pack

that just tags git on a release

jstrachan committed:
fix: avoid ./ on preview

Commit:
Added preview to csharp and updated dotnet version for runtime

jstrachan committed:
Merge pull request #71 from spring-financial-group/master

Added preview to csharp and updated dotnet version for runtime

tomhobson committed:
Update requirements.yaml

jstrachan committed:
Merge pull request #72 from spring-financial-group/master

C sharp Buildpack fix for preview

jstrachan committed:
fix: add a default failureThreashold for knative
